modules/app.xqm

```diff
@@ -177,13 +177,22 @@ declare function app:safe-parse-if-modified-since-header() as xs:dateTime? {
 };
 
 declare function app:last-modified($publication-config as map(*), $document-id as xs:string?, $section-id as xs:string?) {
-    if ($document-id and $section-id and map:contains($publication-config, "section-last-modified")) then (
-        $publication-config?section-last-modified($document-id, $section-id)
-    ) else if ($document-id and map:contains($publication-config, "document-last-modified")) then (
-        $publication-config?document-last-modified($document-id)
-    ) else if (map:contains($publication-config, "publication-last-modified")) then (
-        $publication-config?publication-last-modified()
-    ) else ()
+    let $last-modified-date-time := 
+        if ($document-id and $section-id and map:contains($publication-config, "section-last-modified")) then (
```

Comment:
If there is no "section-last-modified" left in any $publication-config we could drop the entire condition